### DISCUSSION TITLE
Strimzi upgrade problems - 0.29.0 -> 0.32.0

### DISCUSSION THREAD

**Discussion by CROprogrammer** [strimzi/strimzi-kafka-operator#7962] [Q&A]
I have three namespaces (dev, test, prod) on my Openshift, and initially there was Strimzi 0.29.0 installed on each. 
I've decided to upgrade Strimzi to version 0.32.0 in dev namespace, and it worked fine. Later I've changed Kafka users' .yaml files to change deprecated "operation" field to "operations" field (still in dev namespace), and it worked fine.
After that I've tried to upgrade Strimzi to version 0.32.0 in test namespace, and it failed because Kafka user's .yaml file in dev namespace had "operations" field instead of "operation". Then I've reverted changes in Kafka user's .yaml file in dev namespace, and after that Strimzi upgrade in test namespace was successful.
1) Why is that?
2) Do I need to upgrade Strimzi in every namespace before changing Kafka user's .yaml files?

**Comment by scholzj** [comment]
Kubernetes is not multi-tenant. So running multiple versions of an operator is complicated because they all share the same CRD resources. So, when upgrading the operator in the `dev` namespace, you probably upgraded the CRD which will accept the `operations` field in all namespaces now. But the operators in the other namespaces will not understand it until you upgrade them as well. Hope this helps with the understanding.
